lib/transaction_repository.rb

```diff
@@ -106,4 +106,32 @@ def find_all_by_updated_at(updated_at)
   def find_invoice(invoice_id)
     sales_engine.find_invoice_by_transaction_id(invoice_id)
   end
+
+  def charge(payment_data, invoice_id)
+    credit_card_number = payment_data[:credit_card_number]
+    credit_card_expiration_date = payment_data[:credit_card_expiration_date]
+    result = payment_data[:result]
+    new_transaction = Transaction.new(
+        {id: next_id,
```

Comment:
Indent the first parameter one step more than the previous line.<br>Space inside { missing.